cmd/main.go

```diff
@@ -54,8 +56,29 @@ func main() {
 		return
 	}
 
+	// Initialize config from command line
+	config := Configuration{
+		SysFsRoot: sysFsRoot,
+		Mqtt: MqttConfig{
+			Broker: broker,
+			ClientID: clientID,
+			CAFile: caFile,
+			Username: username,
+			Password: password,
+		},
+	}
+
+	// Check if there's a config to be read
+	if configFile != "" {
+		log.Printf("Reading configuration file %s\n", configFile)
+		err := updateConfigFromFile(configFile, &config)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
```

Comment:
This won't work right now, since `Configuration`, `MqttConfig` and `updateConfigFromFile` are only known within the package `unipitt` (`cmd/main.go` is in the `main` package). It's actually the reason the unit tests are broken.

You could just replace `Configuration` and `MqttConfig` by `unipitt.Configuration` and `unipitt.MqttConfig`; `updateConfigFromFile`  would need to be exported (upper case name, i.e. `UpdateConfigFromFile`). I'm not sure I really like that thought, the whole idea of this `Handler` type was that it would be the single entry point into the `unipitt` package, now it's sort of split. You could argue to have something like a configuration handler apart from the main `Handler`, but I wouldn't use more imports than that in `cmd/main.go`

Reply:
Also: preference to use `MQTTConfig` instead of `MqttConfig` (I think it's even a recommended go best practice).

Reply:
I have renamed Mqtt to MQTT now in 96524004e8226cae485bbb0edcfd2827ae85e864

The unit tests were fixed locally, which I have pushed 10 minutes ago and I did exactly what you described: I exported the functions and added qualifications to the types. I am a go newbie after all.

In any case, I argue that the configuration and the command line should somehow get merged _before_ the thing gets to unipitt.Handler, which shouldn't care if an option was populated from the command line or a config file or any other mechanism. I've probably open coded a bit too much in main.go, but I would delegate the processing of the command line and reading of the configuration file where they belong: the main program. Then provide a couple of helper functions for this, so that most of this would be a one-liner in the main program.

Alas, the reading of the configuration file now overrides anything passed from the config file which is not the way it is usually implemented (rather programs initially read the config and command line options override those values). So this could need some improvements.

btw, I extended the config file because I needed yet another option, topic_prefix, which is a string that all mqtt topic names are prefixed with.

Let me know what you would like to do here, for now the unit tests should pass.